src/js/view/controls/components/menu/tizen-menu.js

```diff
@@ -0,0 +1,232 @@
+            const next = isUp ? previousSibling(evt.target) : nextSibling(evt.target);
+            if (next) {
+                next.focus({ preventScroll: true });
+            } else {
+                const nextMenu = isUp ? this.children.captions : this.children.audioTracks;
+                if (nextMenu && !nextMenu.el.contains(evt.target)) {
+                    const items = nextMenu.items;
+                    (isUp ? items[items.length - 1] : items[0]).el.focus({ preventScroll: true });
+                }
+            }
+            this.trigger(USER_ACTION);
+        };
+
+        switch (evt.keyCode) {
+            case 38: // Up
+                onNavigateMenu.call(this, true);
+                break;
+            case 40: // Down
+                onNavigateMenu.call(this, false);
+                break;
+            case 37: // Left
+                this.close();
+                if (this.controlbar) {
+                    this.controlbar.elements.settingsButton.element().focus();
+                }
+                break;
+            case 415: // Play
+            case 10252: // Play/Pause
+                this.close(evt, true);
+                this.api.play({ reason: 'settingsInteraction' });
+                break;
+            case 10009: // Back
+                this.close(evt);
+                break;
+            default:
+                break;
+        }
+    }
+
+    onVisibility(evt) {
+        const el = this.el;
+        
+        if (evt.visible) {
+            this.api.pause({ reason: 'settingsInteraction' });
+            document.addEventListener('keydown', this.onInteraction);
+            el.classList.remove('jw-settings-transition-close');
+            el.classList.add('jw-settings-open');
+            el.classList.add('jw-settings-transition-open');
+            el.setAttribute('aria-expanded', 'true');
+            this.visible = true;
+            const menuNames = Object.keys(this.children);
+            if (this.children && menuNames.length) {
+                // Focus first item of first menu
+                this.children[menuNames[0]].items[0].el.focus({ preventScroll: true });
+            }
+        } else {
+            document.removeEventListener('keydown', this.onInteraction);
+            el.classList.remove('jw-settings-transition-open');
+            el.classList.add('jw-settings-transition-close');
+            el.setAttribute('aria-expanded', 'false');
+            this.visible = false;
+        }
+
+        if (this.controlbar) {
+            // Hide controlbar when showing menu
+            this.controlbar.toggleVisibility(!evt.visible);
+        }
+    }
+
+    open(evt) {
+        if (this.visible) {
+            return;
+        }
+        this.trigger('visibility', { visible: true, evt });
+    }
+
+    close(evt) {
+        if (!this.visible) {
+            return;
+        }
+        this.trigger('visibility', { visible: false, evt });
+    }
```

Comment:
this can be DRYer, it's basically the same implementation
```
open(evt) {
    toggle(evt, true);
}

close(evt) {
    toggle(evt, false);
}

toggle(evt, desiredVisibility) {
    if (this.visible === desiredVisibility) {
        return;
    }
    this.trigger('visibility', { visible: desiredVisibility, evt });
}
```
you would have to change the name though, you already have a toggle function

Reply:
i'd say it's still singular responsibility; the responsibility is to flip the switch